cli/exec/exec.go

```diff
@@ -146,13 +146,17 @@ func execWithAxis(ctx context.Context, c *cli.Command, file, repoPath string, ax
 	}
 
 	environ := metadata.Environ()
-	var secrets []compiler.Secret
-	for key, val := range metadata.Workflow.Matrix {
-		environ[key] = val
-		secrets = append(secrets, compiler.Secret{
-			Name:  key,
-			Value: val,
-		})
+
+	secretMap := metadata.Workflow.Matrix
```

Comment:
I have no idea why `metadata.Workflow.Matrix` is used in cli for secrets in the first place ?!? ... I think I'll add an dedicated option afterwards in an dedicated pull ...